### ISSUE TITLE
I cannot save a PNG image with the alpha channel removed in ImageMagick7.

### ISSUE THREAD

**Issue by naitoh** [rmagick/rmagick#1376]
### Description

I can save PNG images with the alpha channel removed in ImageMagick6, but I cannot save PNG images with the alpha channel removed in ImageMagick7.

### Steps to Reproduce

```sh
$ ruby -v
ruby 3.2.0 (2022-12-25 revision a528908271) [arm64-darwin21]
```

###  ImageMagick 6.9.12-77 : No problem case

```sh
$ /opt/homebrew/Cellar/imagemagick@6/6.9.12-77/bin/compare -version
Version: ImageMagick 6.9.12-77 Q16 aarch64 17704 https://legacy.imagemagick.org
Copyright: (C) 1999 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC Modules 
Delegates (built-in): bzlib fontconfig freetype gslib jng jp2 jpeg lcms ltdl lzma png ps tiff webp xml zlib
```

```ruby
$ irb
irb(main):001:0> require 'rmagick'
=> true
irb(main):002:0> img = Magick::ImageList.new('./png_test_alpha.png')
=> 
[./png_test_alpha.png PNG 552x102 552x102+0+0 DirectClass 8-bit 15kb]
...
irb(main):003:0> img.alpha?
=> true
irb(main):004:0> img.alpha Magick::DeactivateAlphaChannel
=> DeactivateAlphaChannel=4
irb(main):005:0> img.alpha?
=> false
irb(main):006:0> img.write("./test_ok.png")
=> 
[./png_test_alpha.png=>./test_ok.png PNG 552x102 552x102+0+0 DirectClass 8-bit 10kb]
scene=0
irb(main):007:0> img2 = Magick::ImageList.new("./test_ok.png")
=> 
[./test_ok.png PNG 552x102 552x102+0+0 DirectClass 8-bit 10kb]
...
irb(main):008:0> img2.alpha?
=> false
irb(main):009:0> exit
```

In the saved test_ok.png, the alpha channel is removed. 🙆‍♂️ 

* png_test_alpha.png
![png_test_alpha](https://user-images.githubusercontent.com/706220/222936218-36686c03-4445-486c-84cc-ced0b80a2ca7.png)
(It is recommended these images be checked in Github dark mode.)

```sh
$ hexdump png_test_alpha.png |head
0000000 5089 474e 0a0d 0a1a 0000 0d00 4849 5244
0000010 0000 2802 0000 6600 0608 0000 e100 6ac0
0000020 00b0 0400 6924 4343 4950 4343 5020 6f72
0000030 6966 656c 0000 1138 5585 6fdf 54db 3e14
0000040 6f89 a452 3f16 5820 8747 c58a 55af 5b53
0000050 1bb9 ad1a 06c6 9349 eda5 164a e9a5 2ad8
0000060 e424 373a a989 071b e9db aab6 7b4f 3781
0000070 fc06 4001 03d9 480f 213c 060d 7b62 f6d9
0000080 b4c0 5349 2a87 49aa 7b48 c4e8 210f ed26
0000090 5505 bbe1 6276 5327 5cc4 faf5 39cb 39df
```

* test_ok.png
  ![test_ok](https://user-images.githubusercontent.com/706220/222936239-7c3e2915-7e52-4753-857c-1049518af6a1.png)
```sh
$ hexdump test_ok.png |head
0000000 5089 474e 0a0d 0a1a 0000 0d00 4849 5244
0000010 0000 2802 0000 6600 0208 0000 6e00 fda2
0000020 00e7 0300 69e8 4343 6950 6363 0000 8d38
0000030 558d 6fdd 54db 3f14 6f89 a45c 3f16 b1a0
0000040 0e8e 8b15 55af 5b53 1bb9 ad1a 06c6 9349
0000050 e9a5 1a42 cdb9 2ad8 c9a4 6e75 531a 36d7
0000060 d3b6 556d f69f 6f02 f80c 8003 07b2 901e
0000070 4278 0c1a f6c4 edb2 b401 5349 1541 24d5
0000080 3da4 da74 6840 f693 aa82 ae70 53af 5dbb
0000090 b8c6 af91 397f 77e7 ef3e 35d1 c740 9a57
```

### ImageMagick 7.1.0-62 : Problem case

```sh
$ /opt/homebrew/Cellar/imagemagick/7.1.0-62/bin/convert -version
Version: ImageMagick 7.1.0-62 Q16-HDRI aarch64 20885 https://imagemagick.org
Copyright: (C) 1999 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI Modules OpenMP(5.0) 
Delegates (built-in): bzlib fontconfig freetype gslib heic jng jp2 jpeg jxl lcms lqr ltdl lzma openexr png ps raw tiff webp xml zlib
Compiler: gcc (4.2)
```

```ruby
$ irb
irb(main):001:0> require 'rmagick'
=> true
irb(main):002:0> img = Magick::ImageList.new('./png_test_alpha.png')
=> 
[./png_test_alpha.png PNG 552x102 552x102+0+0 DirectClass 8-bit 15kb]
...
irb(main):003:0> img.alpha?
=> true
irb(main):004:0> img.alpha Magick::DeactivateAlphaChannel
=> DeactivateAlphaChannel=5
irb(main):005:0> img.alpha?
=> false
irb(main):006:0> img.write("./test_ng.png")
=> 
[./png_test_alpha.png=>./test_ng.png PNG 552x102 552x102+0+0 DirectClass 8-bit 14kb]
scene=0
irb(main):007:0> img2 = Magick::ImageList.new("./test_ng.png")
=> 
[./test_ng.png PNG 552x102 552x102+0+0 DirectClass 8-bit 14kb]
...
irb(main):008:0> img2.alpha?
=> true
irb(main):009:0> exit
```

In the saved test_ng.png, the alpha channel has not been removed. 🙅‍♂️ 

* test_ng.png
  ![test_ng](https://user-images.githubusercontent.com/706220/222936257-18d400d4-f1f7-4402-9b34-e6230fec703e.png)

```sh
$ hexdump test_ng.png |head
0000000 5089 474e 0a0d 0a1a 0000 0d00 4849 5244
0000010 0000 2802 0000 6600 0608 0000 e100 6ac0
0000020 00b0 0300 69e8 4343 6950 6363 0000 8d38
0000030 558d 6fdd 54db 3f14 6f89 a45c 3f16 b1a0
0000040 0e8e 8b15 55af 5b53 1bb9 ad1a 06c6 9349
0000050 e9a5 1a42 cdb9 2ad8 c9a4 6e75 531a 36d7
0000060 d3b6 556d f69f 6f02 f80c 8003 07b2 901e
0000070 4278 0c1a f6c4 edb2 b401 5349 1541 24d5
0000080 3da4 da74 6840 f693 aa82 ae70 53af 5dbb
0000090 b8c6 af91 397f 77e7 ef3e 35d1 c740 9a57
```

### System Configuration

- ImageMagick version: 7.1.0-62 
- RMagick version: 5.1.0
- Ruby version: 3.2.0
- Environment (Operating system, version and so on): macOS Monterey 12.6.3
- Additional information:

**Comment by dlemstra** [comment]
The `DeactivateAlphaChannel` option works differently in ImageMagick 7. I also don't think you would want to use `DeactivateAlphaChannel` in your situation. It looks like you want to use `OffAlphaChannel` instead but that doesn't seem to be available yet in RMagick. 

I think we should add this to the library where `OffAlphaChannel` and `OnAlphaChannel` should be used directly in IM7 but for IM6 this should be mapped to `DeactivateAlphaChannel` and `ActivateAlphaChannel`. I will open up a PR later today to add support for this.

**Comment by mockdeep** [comment]
@naitoh can you give version 5.2.0 a try? I believe this should be resolved.

**Comment by naitoh** [comment]
I have confirmed that the problem has been resolved in RMagick 5.2.0.
Thanks!
